packages/@aws-cdk/aws-lambda/lib/event-source-filter.ts

```diff
@@ -76,7 +76,19 @@ export class FilterCriteria {
   /**
    * Filter for event source
    */
-  public static filter(filter: {[key:string]: any}): {[key:string]: any} {
-    return { pattern: JSON.stringify(filter) };
+  public static filter(filter: {[key:string]: any}): FilterCriteria {
```

Comment:
this is still a breaking change in the API. I think we do have to deprecate the `filter` property because it will be confusing to have to support two different types of filters in one property.